docs/docs/advanced-use/transient-traits.md

```diff
@@ -0,0 +1,585 @@
+---
+title: Transient Traits and Identities
+---
+
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem';
+
+By default, Flagsmith [stores all Traits](/basic-features/managing-identities#identity-and-trait-storage) associated
+with an Identity to evaluate flags. This default behavior works for most use cases. However, there are scenarios where
+you may want more control over which Traits or Identities are stored. Using the Flagsmith API and SDKs, you can mark
+Traits and Identities as transient — meaning they are used for flag evaluation but are not stored.
+
+## Overview and Use Cases
+
+Transient Identities and Traits are particularly useful when you need to run flag evaluations without storing specific
+data in Flagsmith. Below are common scenarios where transient Traits or Identities come in handy.
+
+### Anonymous Identities
+
+If you choose to provide an empty or blank (`""`) identifier is provided, Flagsmith will automatically treat the
+Identity as transient. In this case, an auto-generated identifier is returned, based on a hash of the provided Traits.
+This ensures consistent flag evaluations without persisting the Identity in the Flagsmith dashboard.
+
+```javascript
+flagsmith.init({
+ evaluationContext: {
+  identity: {
+   identifier: null,
+   traits: { paymentPreference: 'cash' },
+  },
+ },
+});
+let identifierToUseLater = flagsmith.getContext().identity.identifier;
+```
+
+This approach is useful for scenarios such as A/B testing in e-commerce, where you want to include users who haven't
+registered or logged in.
+
+:::info
+
+Server-generated transient identifiers are consistent across flag evaluations, meaning the same identifier will be
+returned for identical sets of Traits.
+
+To generate truly unique identifiers, include additional unique Trait values.
```

Comment:
Do we have an answer to the question "What happens when the user authenticates" yet or not?